From 61503c96491da58bc62cf4dc874d988a12c074a6 Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Sun, 11 Dec 2022 22:56:16 +0100
Subject: [PATCH 29/47] HID: magicmouse: Add .reset_resume for SPI trackpads

The trackpad has to request multi touch reports during resume.

Signed-off-by: Janne Grunau <j@jannau.net>
---
 drivers/hid/hid-magicmouse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hid/hid-magicmouse.c b/drivers/hid/hid-magicmouse.c
index aae3a0f05..6f3df4c99 100644
--- a/drivers/hid/hid-magicmouse.c
+++ b/drivers/hid/hid-magicmouse.c
@@ -1361,7 +1361,7 @@ static int magicmouse_reset_resume(struct hid_device *hdev)
 	 * enable report.  Only the HID_TYPE_USBMOUSE interface accepts it, and
 	 * it must be deferred. Sending it inline here is too early.
 	 */
-	if (msc && hdev->type == HID_TYPE_USBMOUSE)
+	if (msc && (hdev->type == HID_TYPE_USBMOUSE || hdev->bus == BUS_SPI))
 		schedule_delayed_work(&msc->work, msecs_to_jiffies(500));
 
 	return 0;
-- 
2.55.0

